airbyte-integrations/connectors/source-vertica/source_vertica/source.py

```diff
@@ -0,0 +1,173 @@
+                }
+
+            streams.append(
+                AirbyteStream(
+                    name=table,
+                    json_schema={
+                        "$schema": "http://json-schema.org/draft-07/schema#",
+                        "type": "object",
+                        "properties": properties,
+                    }
+                )
+            )
+
+        return AirbyteCatalog(streams=streams)
+
+    def read(
+        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
+    ) -> Generator[AirbyteMessage, None, None]:
+        """
+        Returns a generator of the AirbyteMessages generated by reading the source with the given configuration,
+        catalog, and state.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+            the properties of the spec.yaml file
+        :param catalog: The input catalog is a ConfiguredAirbyteCatalog which is almost the same as AirbyteCatalog
+            returned by discover(), but
+        in addition, it's been configured in the UI! For each particular stream and field, there may have been provided
+        with extra modifications such as: filtering streams and/or columns out, renaming some entities, etc
+        :param state: When a Airbyte reads data from a source, it might need to keep a checkpoint cursor to resume
+            replication in the future from that saved checkpoint.
+            This is the object that is provided with state from previous runs and avoid replicating the entire set of
+            data everytime.
+
+        :return: A generator that produces a stream of AirbyteRecordMessage contained in AirbyteMessage object.
+        """
+        connection = self.create_connection(config)
+
+        for stream in catalog.streams:
+            stream_name = stream.stream.name
+
+            query = f"SELECT * FROM {stream_name}"
+            results, columns = self.query(logger, config, connection, query)
+
+            for row in results:
+                data = dict(zip(columns, row))
+
+                yield AirbyteMessage(
+                    type=Type.RECORD,
+                    record=AirbyteRecordMessage(stream=stream_name, data=data, emitted_at=int(datetime.now().timestamp()) * 1000),
+                )
+
+    def create_connection(self, config: json):
+        return vertica_python.connect(
+            host=config['host'],
+            port=config['port'],
+            user=config['user'],
+            password=config['password'],
+            database=config['database'],
+            schema_name=config['schema_name']
+        )
+
+    def get_tables(self, logger: AirbyteLogger, config: json, connection):
+        query = f'''
```

Comment:
Make sure to filter any system tables

Reply:
`get_tables()` has been implemented to only return the tables that belong to the particular schema we are looking to connect to. Therefore, system tables should be avoided.